### ISSUE TITLE
false positive: unresolved symbol in `core.async/alt!` clauses

### ISSUE THREAD

**Issue by volrath** [borkdude/clj-kondo#393]
**version**

v2019.07.31-alpha

**platform**

Linux

**problem**

Clauses in a [`core.async/alt!`](https://clojuredocs.org/clojure.core.async/alt!) can have a sort of "function body" in which you specify a set of bindings in a vector and a expression that acts on those bindings.

Symbols in these bindings are reported as "unresolved symbol".

**repro**

```
$> echo "(require '[clojure.core.async :as a]) (def c (a/chan)) (a/alt! c ([x] x))" | clj-kondo --lint -
<stdin>:1:68: info: unresolved symbol x
linting took 21ms, errors: 0, warnings: 0
```

**expected behavior**

No info messages.

**Comment by borkdude** [comment]
@volrath If I understand correctly clj-kondo should only do specific analysis for the fn-body like expressions:

```
(alt!
  [c t] ([val ch] (foo ch val)) ;; <- fn-body like expr
  x ([v] v) ;; <- fn-body like expr
  [[out val]] :wrote ;; regular expressions, no special analysis needed
  :default 42)
```

**Comment by borkdude** [comment]
@volrath you can try out this version for testing: https://5244-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2019.08.01-alpha-SNAPSHOT-linux-amd64.zip

**Comment by volrath** [comment]
works for me! you're right about `alt!!` good catch :)

thanks for the super fast fix :+1:

**Comment by borkdude** [comment]
Fixed with https://github.com/borkdude/clj-kondo/commit/73e2c0188f75e9a5a69da70ce37998c3246e6117
